src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRangeGuard.java

```diff
@@ -178,14 +189,38 @@ protected AIState searchTarget()
      */
     protected AIState huntDown()
     {
-        if(huntDownlastAttacker())
+        if (huntDownlastAttacker())
         {
             targetEntity = this.worker.getLastAttacker();
         }
 
+        AbstractEntityBarbarian closestBarbarian = BarbarianUtils.getClosestBarbarianToEntity(worker, currentSearchDistance);
+
+        if (closestBarbarian != null)
+        {
+            targetEntity = closestBarbarian;
+        }
+
+        if (targetEntity != null && worker.getColony() != null)
+        {
+            List<EntityLivingBase> targets = worker.getColony().getGuardTargets();
+            if (targets.stream().noneMatch(entity -> entity == targetEntity))
+            {
+                targets.add(targetEntity);
+            }
+            worker.getColony().setGuardTargets(targets);
+        }
+
         if (!targetEntity.isEntityAlive() || checkForToolOrWeapon(ToolType.BOW))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "targetEntity" is nullable here [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)